ouroboros-consensus/docs/diffusion-pipelining-design/block-diffusion-pipelining.tex

```diff
@@ -0,0 +1,554 @@
+\documentclass[11pt,a4paper]{article}
+
+%include polycode.fmt
+
+\usepackage[margin=2.5cm]{geometry}
+\usepackage{todonotes}
+\usepackage{microtype}
+\usepackage{listings}
+\usepackage{color}
+\usepackage{amssymb,amsmath}
+\usepackage{mathpazo}
+\usepackage{accents}
+\usepackage{longtable,booktabs}
+\usepackage{dcolumn}
+\usepackage{pgf}
+\usetikzlibrary{shapes.geometric}
+\usetikzlibrary{matrix}
+\usepackage{natbib}
+\usepackage{hyperref}
+\usepackage[capitalise,noabbrev,nameinlink]{cleveref}
+\hypersetup{
+  pdftitle={Pipelining in Ouroboros chain diffusion},
+  pdfborder={0 0 0},
+  breaklinks=true
+}
+
+\definecolor{dkgreen}{rgb}{0,0.6,0}
+\definecolor{gray}{rgb}{0.5,0.5,0.5}
+\definecolor{pink}{rgb}{0.92,0.2,0.86}
+
+\lstset{frame=tb,
+  language=Haskell,
+  aboveskip=3mm,
+  belowskip=3mm,
+  showstringspaces=false,
+  columns=flexible,
+  basicstyle={\small\ttfamily},
+  numbers=none,
+  numberstyle=\tiny\color{pink},
+  keywordstyle=\color{pink},
+  commentstyle=\color{pink},
+  stringstyle=\color{pink},
+  breaklines=true,
+  breakatwhitespace=true,
+  tabsize=3
+}
+
+\DeclareMathOperator{\dom}{dom}
+\newcommand\restrict[2]{\left.#1\right||_{#2}}
+\newcommand\deltavar[1]{\accentset{\Delta}{#1}}
+
+\begin{document}
+
+\title {Pipelining in Ouroboros chain diffusion\\
+       {\large \sc An IOHK discussion paper / high level design}}
+\date  {Version 0.4, 19th January 2022}
+\author{Duncan Coutts      \\ {\small \texttt{duncan@well-typed.com}} \\
+                              {\small \texttt{duncan.coutts@iohk.io}} \\
+   \and Matthias Fitzi     \\ {\small \texttt{matthias.fitzi@iohk.io}} \\
+   \and John Woods         \\ {\small \texttt{john.woods@iohk.io}} \\
+                              {\small \texttt{john@postquantum.dev}} \\
+}
+
+\maketitle
+
+\section{Purpose}
+The purpose of this document is to present a proposed change to the algorithm
+for Ouroboros chain diffusion that, if implemented, could facilitate higher block validation times
+for the same choice of the $\Delta$~diffusion time.
+Ergo, we would enjoy a faster $\Delta$~diffusion time in the context of our current block validation requirements.
+This additional budget may be spent on increasing block-size, and/or increasing CPU \& memory limits for Plutus scripts.
```

Comment:
Minor: I think no comma in "block-size, and/or"

Reply:
agreed!